File: android/src/main/java/expo/modules/pluginlocalization/ExpoPluginLocalizationModule.kt
package expo.modules.nativeconfiguration

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import android.content.pm.PackageManager

class ExpoNativeConfigurationModule() : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoPluginLocalization")

    Function("getLocales") {
      val applicationInfo = appContext?.reactContext?.packageManager?.getApplicationInfo(appContext?.reactContext?.packageName.toString(), PackageManager.GET_META_DATA)

      return@Function applicationInfo?.metaData?.getString("LOCALES_SUPPORTED")
    }
  }
}

File: ios/ExpoPluginLocalizationModule.swift
import ExpoModulesCore

public class ExpoNativeConfigurationModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoPluginLocalization")

    Function("getLocales") {
     return Bundle.main.object(forInfoDictionaryKey: "LOCALES_SUPPORTED") as? String
    }
  }
}
